### ISSUE TITLE
Typehint at InvalidReturnStatement incorrectly states fully qualified class names

### ISSUE THREAD

**Issue by HenkPoley** [vimeo/psalm#2654]
I've got an INFO error along the lines of:

```
INFO: InvalidReturnStatement - app/MyNamespace/SomeClass.php:115:16 
- The type 
'array{extras: Illuminate\Support\Collection}' 
does not match the declared return type 
'array{extras: App\MyNamespace\Illuminate\Support\Collection}'
for App\MyNamespace\SomeClass::toArray
```

Which I got after copy-pasting the auto-detected @psalm-return type for that method, from a previous psalm run.

The full qualified class name should be (notice the leading slash) `\Illuminate\Support\Collection` instead of `Illuminate\Support\Collection` or `App\QDT\VO\Illuminate\Support\Collection`. With the leading slash, psalm is happy "No errors found!".

(Additionally, it would be really nice if it could get pretty-printed what the mismatching parts are)

Psalm 3.8.3@389af1bfc739bfdff3f9e3dc7bd6499aee51a831

**Comment by muglug** [comment]
This isn't a bug (at least, I don't consider it one, and PHPStan has the same behaviour). If you want automatic fixes that populate docblocks appropriately, use Psalter: https://psalm.dev/docs/manipulating_code/fixing/#invalidreturntype

**Comment by HenkPoley** [comment]
At the moment `psalter --issues=InvalidReturnType` doesn't quite add @psalm-return docblocks.

**Comment by muglug** [comment]
What _does_ it do in your case?

**Comment by HenkPoley** [comment]
🤔

When I just run `vendor/bin/psalter --dry-run --issues=InvalidReturnType --threads=4` on files where I set something like `@psalm-return foo`, it _will_ impute a psalm return type.

My normal `dry-run-psalter.sh` script uses the following issue fixers string, which does _not_ propose the fixes.

```
--issues=InvalidFalsableReturnType,InvalidNullableReturnType,InvalidReturnType,LessSpecificReturnType,MismatchingDocblockParamType,MismatchingDocblockReturnType,MissingParamType,MissingReturnType
```

**Comment by zmitic** [comment]
I can confirm that ``LessSpecificReturnType`` reports false positives. For a code like this:

```php
class Back implements ConfigurationInterface
{
    public function getAliasName(): string
    {
        return 'back_button';
    }
}
```

and ``./vendor/bin/psalter --dry-run --php-version=7.4 --issues=LessSpecificReturnType``, I get this error:


![image](https://user-images.githubusercontent.com/1964158/80717847-a94b5e00-8af9-11ea-9d25-4826f7ad3a8f.png)

---

So even though the method has typehinted string for return, psalter wants to add phpdoc for it anyways.
